ocaml/xapi/xapi_pool.ml

```diff
@@ -322,7 +322,8 @@ let pre_join_checks ~__context ~rpc ~session_id ~force =
       slavetobe_auth_type slavetobe_auth_service_name ;
     if
       slavetobe_auth_type <> master_auth_type
-      || slavetobe_auth_service_name <> master_auth_service_name
+      || String.lowercase_ascii slavetobe_auth_service_name
```

Comment:
Slight simplification:
```
String.(lowercase_ascii slavetobe_auth_service_name <> lowercase_ascii master_auth_service_name)
```
but fine as is.

Reply:
As discussed in person, this does not work for UTF8 but it still covers the common case to get over the hurdle that someone is using no all lowercase to begin with. We need to be sure that it is indeed to compare these names as lowercase such that `foo` and `FOO` are indeed semantically equal and that we don't make names equal that in fact designate different entities.

Reply:
Reading Microsoft specifications domain names come from Netbios names and/or DNS names which are both defined in terms of ASCII. See https://learn.microsoft.com/en-us/troubleshoot/windows-server/active-directory/naming-conventions-for-computer-domain-site-ou although you need to refer also to Netbios/DNS specifications.